crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -851,6 +856,52 @@ mod tests {
         handle.await.unwrap();
     }
 
+    #[test]
+    fn test_peer_lower_capability_version() {
+        let local_capabilities: Vec<Capability> = vec![EthVersion::Eth66.into(), EthVersion::Eth67.into()];
```

Comment:
Should we just use a HashSet instead of Vec capability and use set intersection methods? I think I don't understand why we hand rolled this cc @Rjected @mattsse

Reply:
this follows the spec, caps are a list.
It's unclear whether entries must be unique or sorted.

there's also no rlp impls for (Hash|Btree)Set currently.

> I don't understand why we hand rolled this

hand rolled what exactly?

Reply:
We can convert them to a set internally then convert back?

I mean handrolling the intersection logic

Reply:
yeh, we can probably simplify the matching by a lot by using `HashSet::intersection`, maybe collected into BtreeSet, then select the minimum versions

Reply:
Yeah that's what I had in mind.

Reply:
Haven't looked at the implementation of `intersection` but I figure it does what we do here: iterate over one set, and for each element, check if the other `contains` it.

Hand-rolling this here lets us only keep the maximum version in the intersection while only using one iteration.

Likely a small optimization to be fair

Reply:
right, this way we get rid of the dedupe step and don't need to tmp allocate a set